flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -1122,6 +1178,35 @@ void createTemporarySystemFunction(
     /** Gets the names of all functions in this environment. */
     String[] listFunctions();
 
+    /**
+     * Gets the names of all models available in the given namespace (the given database of the
+     * given catalog). It returns both temporary and permanent models.
+     *
+     * @return A list of the names of all registered models in the given database of the given
+     *     catalog.
+     * @see #listTemporaryModels()
+     */
+    String[] listModels();
+
+    /**
+     * Gets the names of all models available in the given namespace (for the provided database of
```

Comment:
nit: I would not use the word namespace- I would refer to the catalog and database names 
  please add @params